bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -288,21 +288,24 @@ private void startConsole(Console console, String[] args) {
         final String consoleScript = script.contains("/") ? script : DEFAULT_CONSOLE_PATH + script;
         final String[] argv = args;
 
-        // Use full class name here to avoid ambiguity against other OSGiConsole implementations
-        if (console instanceof org.openhab.core.io.console.karaf.OSGiConsole) {
-            logger.debug("Starting JRuby console with script: {}", consoleScript);
-
-            executeWithFullJRuby(console, engine -> {
-                // Resolve console.getSession().getTerminal() in Ruby to avoid having to add
-                // org.apache.karaf.shell.core as a dependency in pom.xml
-                engine.put("$console", console);
-                engine.put(ScriptEngine.ARGV, argv);
-                engine.eval(String.format("$terminal = $console.session.terminal; require '%s'", consoleScript));
-                return null;
-            });
-        } else {
-            console.println("JRuby Console is not supported in this environment.");
-        }
+        logger.debug("Starting JRuby console with script: {}", consoleScript);
+
+        executeWithFullJRuby(console, engine -> {
+            // Resolve console.getSession().getTerminal() in Ruby to avoid having to add
+            // org.apache.karaf.shell.core as a dependency in pom.xml
+            engine.put("$console", console);
+            engine.put(ScriptEngine.ARGV, argv);
+            engine.eval("""
+                    begin
+                      $terminal = $console.session.terminal
+                    rescue NoMethodError
+                      puts "Error: Unable to resolve terminal. JRuby Console is not supported in this environment."
+                      exit
+                    end
```

Comment:
Should we just put all of this into console.rb?

Reply:
yes